source/concepts/protocols.rst

```diff
@@ -0,0 +1,309 @@
+2. the speed of communication channels used for exchanging states & signatures.
+
+If a participant knows an update to be proposed is the final off-chain
+transaction, then it should be marked as ``final``. This is not a mandatory
+part. If it is not done, then a separate off-chain transaction that marks the
+latest off-chain state as final will be sent in the next phase.
+
+.. image:: ../_generated/concepts/transact_generic.svg
+  :align: Center
+  :alt: Image not available
+
+Register phase
+--------------
+
+In this phase, the state of the channel which should be settled will be agreed
+upon. The balances locked in the parent layer will be redistributed to the
+participants as per the balances in the settled state.
+
+This phase consists of three sub-phases:
+
+1. Finalize
+2. Register dispute on-chain
+3. Progress channel on-chain
+
+1. Finalize
+```````````
+
+In this sub-phase, the protocol tries to create an agreement on the state to be
+settled without any on-chain transactions. If it succeeds, then the channel
+direclty progresses to the ``settle phase``. If not, then it transitions to the
+next sub-phase. This protocol is same for all three types of channels.
+
+If the latest off-chain state was marked as ``final`` already in the transact
+phase, it implies both the participants have agreed it to be the state to be
+settled. If it was not, another another off-chain transaction is sent, on behalf
+of the participant who initiated the channel settlement, marking the latest
+off-chain state as ``final``. It is up to the other participant to accept or
+reject or ignore it.
+
+If accepted, channel transitions to ``settle phase``, if not it transitions to
+the next sub-phase.
+
+.. image:: ../_generated/concepts/register_generic_finalize.svg
+  :align: Center
+  :alt: Image not available
+
+
+2. Register dispute on-chain
+````````````````````````````
+
+In this sub-phase, the protocol creates an agreement on the state to be
+settled by registering a dispute with the parent layer and resolving it.
+
+The parent layer for registering dispute is always the blockchain. In case of
+
+1. Ledger channel: The state of the channel, all its sub-channels and virtual
+   channels must be collected and registered on the blockchain.
+
+2. Sub-channel: The state of the parent channel, all the sub-channels and
+   virtual channels of the parent channel must be collected and registered on
+   the blockchain.
+
+3. Virtual channel: The state of the parent channel between the participant who
+   initiated channel settlement and the intermediary, all the sub-channels and
+   virtual channels of this parent channel must be collected and registered on
+   the blockchain.
+
+   Once the common intermediary is notified that one of the parent channels
+   holding funds for the virtual channel has been registered on-chain for
+   dispute resolution, the intermediary will try to finalize the state of the
```

Comment:
Currently, the intermediary does not immediately try to finalize the state with the other participant. It simply stores the newest virtual channel state, so that it can potentially register and conclude the channel tree with the other participant.

Reply:
Yes, I think what I have described is incorrect. I got clarified when I was thinking through, which I have also added in this [comment](https://github.com/hyperledger-labs/perun-proposals/pull/11#issuecomment-898348475) in proposals repository for watcher design.

Updated the description.